exonum-light-client/src/main/java/com/exonum/client/response/BlocksResponse.java

```diff
@@ -28,12 +28,12 @@
   List<Block> blocks;
 
   /**
-   * The smallest height of the returned blocks that match the search criteria.
+   * The smallest inclusive height of the returned blocks that match the search criteria.
    */
   long blocksRangeStart;
 
   /**
-   * The largest height of the returned blocks that match the search criteria.
+   * The largest exclusive height of the returned blocks that match the search criteria.
    * If some blocks in the range do not match the search criteria then:
    * {@code blocksRangeEnd - blocksRangeStart != blocks.size}.
```

Comment:
The blocks API is rather confusing. If they don't match (i.e., skip empty blocks), then what do I get in `end` — `latest + 1` or `maxMatching + 1`? If the first, then why isn't it `lastRequested = end-1`, because it doesn't have a use as `end` as you won't use it in iteration. If the second — then again, why on Earth anyone needs that — just take blocks[-1].height and you are fine (if there is a block, obviously, which, again, indicates that we don't need 'maxMatching + 1').

CC @aleksuss

Reply:
@dmitry-timofeev ok. I'll fix it.

Reply:
Thanks! Which names do you think would be clear on this? firstInRange + latestRequested? firstInRange + lastInRange? first + last?

Reply:
I suppose `first + last` would be clear enough.

Reply:
Yes, with appropriate documentation, warning that `last` is the last requested (= `latest`, if specified, or current blockchain height, if omitted), not necessarily the height of the last block in the returned list, because some (empty) might have been filtered.